packages/connection/index.js

```diff
@@ -106,49 +107,50 @@ class Connection extends EventEmitter {
     return this.jid
   }
 
-  _online() {
-    this.emit('online', this.jid)
-  }
-
-  _authenticated() {
-    this.emit('authenticated')
+  _status(status, ...args) {
+    this.status = status
+    this.emit('status', status, ...args)
+    this.emit(status, ...args)
```

Comment:
So in case of client going offline I need to start a new connection, which is a completely different session, and I should request a new 'online' promise when starting it.